sky/tools/create_macos_framework.py

```diff
@@ -143,108 +130,78 @@ def regenerate_symlinks(fat_framework_bundle):
   )
 
 
-def embed_codesign_configuration(config_path, content):
-  with open(config_path, 'w') as file:
-    file.write(content)
-
-
 def process_framework(dst, args, fat_framework_bundle, fat_framework_binary):
   if args.dsym:
     dsym_out = os.path.splitext(fat_framework_bundle)[0] + '.dSYM'
-    subprocess.check_call([DSYMUTIL, '-o', dsym_out, fat_framework_binary])
+    sky_utils.extract_dsym(fat_framework_binary, dsym_out)
     if args.zip:
       dsym_dst = os.path.join(dst, 'FlutterMacOS.dSYM')
-      subprocess.check_call(['zip', '-r', '-y', 'FlutterMacOS.dSYM.zip', '.'], cwd=dsym_dst)
+      sky_utils.create_zip(dsym_dst, 'FlutterMacOS.dSYM.zip', ['.'], symlinks=True)
       # Double zip to make it consistent with legacy artifacts.
       # TODO(fujino): remove this once https://github.com/flutter/flutter/issues/125067 is resolved
-      subprocess.check_call([
-          'zip',
-          '-y',
-          'FlutterMacOS.dSYM_.zip',
-          'FlutterMacOS.dSYM.zip',
-      ],
-                            cwd=dsym_dst)
-      # Use doubled zipped file.
+      sky_utils.create_zip(dsym_dst, 'FlutterMacOS.dSYM_.zip', ['FlutterMacOS.dSYM.zip'])
+
+      # Overwrite the FlutterMacOS.dSYM.zip with the double-zipped archive.
       dsym_final_src_path = os.path.join(dsym_dst, 'FlutterMacOS.dSYM_.zip')
       dsym_final_dst_path = os.path.join(dst, 'FlutterMacOS.dSYM.zip')
       shutil.move(dsym_final_src_path, dsym_final_dst_path)
 
   if args.strip:
-    # copy unstripped
     unstripped_out = os.path.join(dst, 'FlutterMacOS.unstripped')
-    shutil.copyfile(fat_framework_binary, unstripped_out)
+    sky_utils.strip_binary(fat_framework_binary, unstripped_out)
+
+
+def zip_framework(dst):
+  framework_dst = os.path.join(dst, 'FlutterMacOS.framework')
+  sky_utils.write_codesign_config(os.path.join(framework_dst, 'entitlements.txt'), [])
+  sky_utils.write_codesign_config(
+      os.path.join(framework_dst, 'without_entitlements.txt'),
+      [
+          # TODO(xilaizhang): Remove the zip file from the path when outer zip is removed.
```

Comment:
nit: Xilai is probably not going to do this =(

Reply:
sgtm -- let me think of a name that isn't Xilai or me who also won't do this. 😂

Reply:
Alright, I have accepted the heavy burden of another TODO that I probably won't do.